include/rmm/mr/device/detail/stream_ordered_memory_resource.hpp

```diff
@@ -201,7 +200,7 @@ class stream_ordered_memory_resource : public crtp<PoolResource>, public device_
    */
   void* do_allocate(std::size_t size, cuda_stream_view stream) override
   {
-    RMM_LOG_TRACE("[A][stream {:p}][{}B]", fmt::ptr(stream.value()), size);
+    RMM_LOG_TRACE("[A][stream %s][%dB]", rmm::detail::format_stream(stream), size);
```

Comment:
*nit*: `format_stream` produces a decimal representation. Is there a reason we couldn't just use:

```
RMM_LOG_TRACE("[A][stream %p][%dB], stream.value(), size);
```
?

Reply:
`%p` outputs zero as the string "(nil)", which I found out by failing `logger_tests.cpp`.  I don't think we want the null stream to be printed as "(nil)".

Reply:
Appears that %p formatting is implementation defined. https://en.cppreference.com/w/c/io/fprintf

Reply:
Ah, that is a good reason not to use `%p`